xtask/src/release.rs

```diff
@@ -0,0 +1,154 @@
+use serde_json::json;
+use xshell::{cmd, read_file};
+
+use crate::{config, flags, project_root, Result};
+
+impl flags::Release {
+    /// Run the release command to effectively create a release.
+    pub(crate) fn run(self) -> Result<()> {
+        let config = self.get_config()?;
+
+        let version = self.get_version()?;
+        println!("Making GitHub release for {} {}…", self.name, version);
+
+        let changes = self.get_changes(&version)?;
+
+        let tag = format!("{}-{}", self.name, version);
+        let name = format!("{} {}", self.name, version);
+
+        cmd!("git tag").args(&["-a", &tag, "-m", &name, "-m", &changes]).secret(true).run()?;
+
+        cmd!("git push").args(&[&config.remote, &tag]).run()?;
+
+        cmd!("curl")
+            .args(&[
+                "-u",
+                &format!("{}:{}", config.user, config.token),
+                "-X",
+                "POST",
+                "-H",
+                "Accept: application/vnd.github.v3+json",
+                &format!("https://api.github.com/repos/{}/releases", config.repo),
+                "-d",
+                &json!({
+                    "tag_name": tag,
+                    "name": name,
+                    "body": changes,
```

Comment:
One thing that has been annoying me is GitHub treating newlines as newlines in the output, rather than requiring two spaces at the end of the line to create a newline in the HTML output. It's not that important, but if you could scan the changes and transform a newline followed by a non-newline character into a space, for GitHub only, that would be great.